protoc-gen-openapiv2/internal/genopenapi/template.go

```diff
@@ -1282,9 +1281,18 @@ func renderServices(services []*descriptor.Service, paths *openapiPathsObject, r
 								}
 								desc = messageSchema.Description
 							} else {
-								schema = messageSchema
-								if schema.Properties == nil || len(*schema.Properties) == 0 {
-									grpclog.Warningf("created a body with 0 properties in the message, this might be unintended: %s", *meth.RequestType)
+								if messageSchema.Properties != nil && len(*messageSchema.Properties) != 0 {
+									strategyFn := LookupNamingStrategy(reg.GetOpenAPINamingStrategy())
+									if strategyFn == nil {
+										panic(fmt.Errorf("could not find OpenAPI naming lookup naming strategy '%s'", reg.GetOpenAPINamingStrategy()))
+									}
+									strats := strategyFn([]string{meth.FQMN(), svc.FQSN()})
+									defName := fmt.Sprintf("%sBody", strats[meth.FQMN()])
```

Comment:
I'm afraid I don't think this will be sufficient. Services are unique per file, but we support merging files into a single swagger file. I think we'll need to keep track of names and just add something to the end when we discover a collision. It'd need to be carried across all file generations so it'll be a bit more invasive than this. What do you think?

Reply:
I changed to use the same mechanism as for messages: the FQN of the method coupled with the `fullyQualifiedNameToOpenAPIName` function. The function will generate the smallest non-ambiguous FQN, based on the naming strategy.

Reply:
Sounds great, thanks